examples/sha2-chain/src/main.rs

```diff
@@ -1,11 +1,28 @@
+use sha2::{Digest, Sha256};
+
+fn compute_native(input: [u8; 32], num_iters: u32) -> [u8; 32] {
+    let mut hash = input;
+    for _ in 0..num_iters {
+        let mut hasher = Sha256::new();
+        hasher.update(&hash);
+        let res = &hasher.finalize();
+        hash = Into::<[u8; 32]>::into(*res);
+    }
+
+    hash
+}
+
 pub fn main() {
     let (prove_sha2_chain, verify_sha2_chain) = guest::build_sha2_chain();
 
     let input = [5u8; 32];
     let iters = 100;
+    let native_output = compute_native(input, iters);
```

Comment:
You should be able to run `guest::sha2_chain` which will support native computation of the function rather than rewriting.